src/Middleware/HttpLogging/src/HttpLoggingMiddleware.cs

```diff
@@ -129,9 +156,11 @@ private async Task InvokeInternal(HttpContext context)
                 }
             }
 
-            var httpRequestLog = new HttpRequestLog(list);
+            var httpRequestLog = new HttpRequestLog(logContext.Parameters);
 
             _logger.RequestLog(httpRequestLog);
+
+            logContext.Parameters.Clear();
```

Comment:
This is the only thing we're resetting between `OnRequestAsync` and `OnResponseAsync`. That is very unclear right now. I can modify what response fields are going to be logged during `OnRequestAsync` but I can't add arbitrary metadata to be logged for both request and response.

We should be very clear in the documentation and possibly reset all the properties on the context (although based on the tests it's intentional to be able to modify response flags).

Reply:
> We should be very clear in the documentation and possibly reset all the properties on the context (although based on the tests it's intentional to be able to modify response flags).

Yes, being able to control the response log settings during the request callback is intentional, it lets people avoid duplicate checks.

I'll work on the docs.